packages/react-app/src/components/FromToken.jsx

```diff
@@ -47,11 +47,13 @@ export const FromToken = () => {
   const fallbackLogo = isxDaiChain(network.value) ? xDAILogo : EthLogo;
 
   useEffect(() => {
-    if (token && account) {
+    if (!networkMismatch && token && account && ethersProvider) {
       setBalance();
-      fetchTokenBalance(token, account).then(b => setBalance(b));
+      fetchTokenBalanceWithProvider(ethersProvider, token, account).then(b =>
+        setBalance(b),
+      );
     }
-  }, [token, account, setBalance]);
+  }, [token, account, setBalance, ethersProvider, networkMismatch]);
```

Comment:
When switching networks after the app as loaded and before you switch wallets, for some reason `networkMismatch` is getting set AFTER this useEffect gets called. causing the error below. not sure why this is happening

<img width="1623" alt="Screen Shot 2020-10-16 at 5 41 08 PM" src="https://user-images.githubusercontent.com/5998100/96322738-89d8bc80-0fd7-11eb-9746-2c4392360141.png">

Reply:
networkMismatch gets set in a useEffect itself if I'm not wrong.. 
if we can't get around it then maybe this one call we could still do using the infura provider.